application/src/main/java/org/thingsboard/server/actors/ruleChain/RuleNodeActorMessageProcessor.java

```diff
@@ -160,4 +179,22 @@ private TbNode initComponent(RuleNode ruleNode) throws Exception {
     protected RuleNodeException getInactiveException() {
         return new RuleNodeException("Rule Node is not active! Failed to initialize.", ruleChainName, ruleNode);
     }
+
+    private boolean isMySingletonNode() {
+        return !ruleNode.isSingletonMode()
+                || systemContext.getDiscoveryService().isMonolith()
+                || defaultCtx.isLocalEntity(ruleNode.getId());
```

Comment:
local entity relay on `ruleNodeId`
`putToQueue` does not use the `ruleNodeId`, so desired partition is unreachable.

Reply:
`TopicPartitionInfo tpi = systemContext.resolve(ServiceType.TB_RULE_ENGINE, tbMsg.getQueueName(), tenantId, entityId);`

entityId it is ruleNodeId